graf2d/gpad/src/TPad.cxx

```diff
@@ -653,7 +653,7 @@ void TPad::Clear(Option_t *option)
       getchar();
    }
 
-   if (!gPad->IsBatch()) GetPainter()->ClearDrawable();
+   if (!gPad->IsBatch()&&GetPainter()) GetPainter()->ClearDrawable();
    if (gVirtualPS && gPad == gPad->GetCanvas()) gVirtualPS->NewPage();
```

Comment:
These protections look fine. One little cosmetic thing. we usually put space around `&&` i.e.:
```
if (!gPad->IsBatch() && GetPainter()) ... 
```

Reply:
Fixed.